src/twisted/protocols/test/test_tls.py

```diff
@@ -977,6 +985,58 @@ def disconnected(ign):
         return disconnectDeferred
 
 
+    def test_noCircularReferences(self):
+        """
+        TLSMemoryBIOProtocol doesn't leave circular references that keep
+        it in memory after connection is closed.
+        """
+        def nObjectsOfType(type):
+            """
+            Return the number of instances of a given type in memory.
+
+            @param type: Type whose instances to find.
+
+            @return: The number of instances found.
+            """
+            return sum(1 for x in gc.get_objects() if isinstance(x, type))
+
+        gc.disable()
```

Comment:
fwiw you can get rid of the whole `try/finally` with `self.addCleanup(gc.enable)` right here.

Reply:
Thanks, I forgot about it